artiq/tools.py

```diff
@@ -20,7 +20,8 @@
 __all__ = ["parse_arguments", "elide", "short_format", "file_import",
            "get_experiment",
            "exc_to_warning", "asyncio_wait_or_cancel",
-           "get_windows_drives", "get_user_config_dir"]
+           "get_windows_drives", "get_user_config_dir",
+           "dataset_to_string"]
```

Comment:
Will try to find a better place for it. Placed it in `gui/tools.py` earlier but as you mentioned it needs to be accessible in `artiq_client`.  Will try to find a better place for it.

Reply:
Considering the `dataset_to_string` function is a replacement / alternative to the `short_format` function in `dashboard/datasets` (and also in `frontend/artiq_client` and `dashboard/browser` in future PRs) I have placed it right after it in `tools.py` for now. Looking at some alternative locations for the function but considering it will be used in the same files as `short_format` it may have to stay in `tools.py`